ambry-api/src/main/java/com.github.ambry/network/NetworkSendMetrics.java

```diff
@@ -0,0 +1,65 @@
+/**
+ * Copyright 2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.network;
+
+import com.codahale.metrics.Histogram;
+
+
+/**
+ * Tracks a set of metrics for a {@link NetworkSend}
+ */
+public class NetworkSendMetrics {
+  private long timeSpentTillNow;
+  private Histogram requestOrResponseQueueTime;
+  private Histogram requestOrResponseSendTime;
+  private Histogram requestOrResponseTotalTime;
+  private Histogram requestOrResponseSendTimeBySize;
+  private Histogram requestOrResponseTotalTimeBySize;
+
+  public NetworkSendMetrics(Histogram requestOrResponseQueueTime, Histogram requestOrResponseSendTime,
+      Histogram requestOrResponseTotalTime, Histogram requestOrResponseSendTimeBySize,
+      Histogram requestOrResponseTotalTimeBySize, long timeSpentTillNow) {
+    this.requestOrResponseQueueTime = requestOrResponseQueueTime;
+    this.requestOrResponseSendTime = requestOrResponseSendTime;
+    this.requestOrResponseTotalTime = requestOrResponseTotalTime;
+    this.timeSpentTillNow = timeSpentTillNow;
+    this.requestOrResponseSendTimeBySize = requestOrResponseSendTimeBySize;
+    this.requestOrResponseTotalTimeBySize = requestOrResponseTotalTimeBySize;
+  }
+
+  /**
+   * Updates the time spent by the request or response in the queue before being sent out
+   * @param value the time spent by the request or response in the queue
+   */
+  public void updateQueueTime(long value) {
```

Comment:
I felt the reasoning as follows. NetworkSend (which encapsulates "Send") is an object which has to be sent across the network. So, irrespective of server or client (or even any other consumer) will queue it up somewhere and then starts to send the "NetworkSend". Selector might take N no of polls(or writes) to complete the actual send. Hence we have 3 metrics, queueTime, sendTime and totalTime.